### ISSUE TITLE
Make compatible with Strapi 5

### ISSUE THREAD

**Issue by netzpixel** [mattmilburn/strapi-plugin-menus#164]
It seams the plugin is not yet compatible with version 5 of strapi.
I just tried to upgrade a v4 project and got this error:

`Error: [vite]: Rollup failed to resolve import "@strapi/helper-plugin"     ││   from "/opt/node_modules/strapi-plugin-menus/admin/src/index.js".`

**Comment by jwalcher** [comment]
any news on this? anxious to migrate to strapi 5...

**Comment by bennyhoyer** [comment]
also waiting for a new strapi 5 release.
has anyone tried #157 with strapi 5? seems like this pr is very old and is maybe missing something but interesting if it already works.

**Comment by rihards-simanovics** [comment]
@mattmilburn is there any timeline for compatibility with v5?

**Comment by rihards-simanovics** [comment]
Hey @mattmilburn, I understand you are busy, and this is a free project which you do in free time, but I do need to know if this menu plugin is going to work with Strapi v5 in the next months or so, as I'm amidst redeveloping a client's website and they need the site ready before new year.

I'm not pulling people away from here but if you re looking for something that will work in the interim whiles the project is catching up with Strapi V5 anyone who is waiting for this project, you may want to check out this plugin it's called tree-menus: https://market.strapi.io/plugins/tree-menus. Just be aware there is a hot fix as there is a small issue with displaying the menu items builder, see the notice under [installation step](https://market.strapi.io/plugins/tree-menus#installation).

**Comment by mattmilburn** [comment]
Hi @rihards-simanovics This plugin probably won't be upgraded in the next month. I manage 5 plugins right now and some of the simpler ones are having issues migrating to v5 and I'm still waiting on Strapi to respond to the issue I posted about it. Sorry, I wish I could speed up the process but Menus is considerably larger and more complex compared to my other plugins and most likely will undergo an exhaustive refactoring so I've just been saving it for last.

**Comment by rihards-simanovics** [comment]
> I wish I could speed up the process but Menus is considerably larger and more complex compared to my other plugins and most likely will undergo an exhaustive refactoring so I've just been saving it for last.

No worries, I understand, thanks for the update. The plugin I've posted in my previous comment, works with V5, perhaps you could use some code from there. Anyways, thanks for the update.

**Comment by jwalcher** [comment]
Coming back to this... 'tree-menus' mentioned by @rihards-simanovics seems to work fine with the hot fix, and provides essentially the same functionality. I still prefer 'menus' (as it works better with my localization scheme), so I'm happy to wait some more. @mattmilburn, do you definitely know that it will be migrated to Strapi 5?

**Comment by mattmilburn** [comment]
@jwalcher Yes, this plugin is still in development for v5. It may take some time as I do have a couple other plugins that are being worked on first.

**Comment by ricag-dev** [comment]
I used vite.config.js 

```import { mergeConfig } from "vite";

export default (config) => {
  // Important: always return the modified config
  return mergeConfig(config, {
    resolve: {
      alias: {
        "@": "/src",
        "@strapi/helper-plugin": "@strapi/strapi/admin",
        "@strapi/icons/Check": "/node_modules/@strapi/icons/dist/icons/Check.d.ts",
        
      },
    },
  });
};```

**Comment by jwalcher** [comment]
@mattmilburn yet another friendly inquiry about v5 compatibility of menus plugin. I'd really like to migrate to Strapi 5 (and NextJS 15 in tow) at some point :)

**Comment by gosi-at** [comment]
Same here, thanks for providing this plugin, is there any rough ETA on when it will happen?

**Comment by joe69** [comment]
@mattmilburn yet another friendly inquiry about v5 compatibility of menus plugin.

**Comment by jakubjanousek** [comment]
> I used vite.config.js
> 
> ```
> 
> export default (config) => {
>   // Important: always return the modified config
>   return mergeConfig(config, {
>     resolve: {
>       alias: {
>         "@": "/src",
>         "@strapi/helper-plugin": "@strapi/strapi/admin",
>         "@strapi/icons/Check": "/node_modules/@strapi/icons/dist/icons/Check.d.ts",
>         
>       },
>     },
>   });
> };```
> ```

@ricag-dev  does it mean the plugin works for you with this config?
I still get the error

```
  [commonjs--resolver] Expression expected in /Users/.../strapi/node_modules/strapi-plugin-menus/admin/src/pages/App/index.js
file: /.../node_modules/strapi-plugin-menus/admin/src/pages/App/index.js:21:4
19: const App = () => { 
20:   return ( 
21:     <QueryClientProvider client={queryClient}>
 ^    
22:       <Layout> 
23:         <Switch>   
```

**Comment by shuaibkhandev** [comment]
Could you please Confirm whether this plugin is currently compatible with Strapi v5, or if it still does not suport Strapi v5?

**Comment by gosi-at** [comment]
> Could you please Confirm whether this plugin is currently compatible with Strapi v5, or if it still does not suport Strapi v5?

there is no update for v5, you can try another plugin but also not maintained for a logn time: https://market.strapi.io/plugins/tree-menus
